plugins/UsersManager/templates/userSettings.twig

```diff
@@ -107,6 +107,15 @@
     </form>
 </div>
 
+<div piwik-content-block
+     content-title="{{ 'UsersManager_TokenAuth'|translate|e('html_attr') }}">
+    <p id="token_auth_user" data-token="{{ userTokenAuth }}">{{ userTokenAuth }}</p>
+
+    <button class="btn btn-link"
+            ng-controller="PersonalSettingsController as personalSettings"
+            ng-click="personalSettings.regenerateTokenAuth()">{{ 'UsersManager_RegenerateTitle'|translate }}</button>
```

Comment:
Maybe we could also add something like "Once the token has been regenerated don't forget to update the token in case you are using the token to export reports via the reporting API, to track data or to embed widgets"